src/app/index.route.coffee

```diff
@@ -160,7 +189,13 @@ angular.module 'mnoEnterpriseAngular'
           controllerAs: 'vm'
 
     $urlRouterProvider.otherwise ($injector, $location) ->
-      unless $injector.get('MnoeConfig').isOnboardingWizardEnabled()
+      MnoeConfig = $injector.get('MnoeConfig')
+      MnoeCurrentUser = $injector.get('MnoeCurrentUser')
+
+      if !MnoeCurrentUser.get() && MnoeConfig.arePublicPagesEnabled()
```

Comment:
If you are on a page that is not public, a promise is returned. But I changed the method a bit. In the case that you are on a public page AND a user is not logged in, it returns null. I did this for routing purposes as `MnoeCurrentUser.get()` is called in `index.run.coffee` so if for some reason you are on a public product page and hit refresh it will actually just refresh the page. Previously, it would always redirect you back to the landing page instead of allowing you to hit refresh on a product page.